src/kernels/jupyter/finder/remoteKernelFinder.ts

```diff
@@ -55,6 +67,18 @@ export class RemoteKernelFinder implements IRemoteKernelFinder, IExtensionSingle
     kind: string = 'remote';
     private cache: RemoteKernelConnectionMetadata[] = [];
 
+    private _onDidChangeKernels = new EventEmitter<void>();
+    onDidChangeKernels: Event<void> = this._onDidChangeKernels.event;
+
+    private _initializeResolve: () => void;
+    private _initializedPromise: Promise<void>;
+
+    get initialized(): Promise<void> {
+        return this._initializedPromise;
+    }
+
+    private wasPythonInstalledWhenFetchingControllers = false;
```

Comment:
Minor, but might be nice to have a name change on this (it's not fetching controllers).